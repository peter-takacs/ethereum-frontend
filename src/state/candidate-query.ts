
export interface CandidateQuery {
    candidate?: string,
    certificate?: string
}