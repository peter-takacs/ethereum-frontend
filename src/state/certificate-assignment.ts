export enum CertificateQueryState {
    Idle,
    Loading,
    Assigned,
    NotAssigned
}

export interface State {
    candidate: string,
    certificate: string,
    state: CertificateQueryState
}