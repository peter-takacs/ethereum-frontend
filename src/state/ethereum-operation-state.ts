export enum EthereumOperationState {
    Idle,
    InProgress,
    Completed,
    Error
}