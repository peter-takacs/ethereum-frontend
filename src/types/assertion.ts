import { Address } from "./ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}