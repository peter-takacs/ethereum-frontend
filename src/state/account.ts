import { Address } from "../types/ethereum-address";

export interface AccountState {
    address: Address | null;
}