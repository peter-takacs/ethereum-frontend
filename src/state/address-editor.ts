import { Address } from "../types/ethereum-address";

export interface State {
    placeholder?: string;
    address: Address | null
}