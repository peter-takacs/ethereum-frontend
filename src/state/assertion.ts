import { Address } from "../types/ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}