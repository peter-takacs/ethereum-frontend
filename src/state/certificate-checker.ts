import { Address } from "../types/ethereum-address";

export enum CertificateCheckState {
    Idle,
    Checking,
    Confirmed,
    Rejected
}
export interface State {
    candidate: Address | null,
    certificate: string,
    certificateStatus: CertificateCheckState
}