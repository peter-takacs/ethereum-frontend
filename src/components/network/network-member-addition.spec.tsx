import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CertificateChecker from '../certificates/certificate-checker';
import { dummyAddress } from '../../utils/test/address';
import CertificateLister from '../certificates/certificate-lister';
import { EthereumOperationState } from '../../state/ethereum-operation-state';
import { Assertion } from '../../types/assertion';
import NetworkMemberAddition from './network-member-addition';

test('Renders correctly in every state without results', () => {
    [
        EthereumOperationState.Idle,
        EthereumOperationState.Completed,
        EthereumOperationState.Completed,
        EthereumOperationState.Error
    ].forEach(state => {
        const component = renderer.create(
            <NetworkMemberAddition
                onSubmit={() => { }}
                status={state}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})

test('Renders correctly with invalid address', () => {
    const component = renderer.create(
        <NetworkMemberAddition
            onSubmit={() => { }}
            status={EthereumOperationState.Completed}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

