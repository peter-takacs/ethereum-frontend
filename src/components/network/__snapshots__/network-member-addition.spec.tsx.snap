// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly in every state without results 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiFormControl-root-98"
  >
    <div
      className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-110"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Candidate to be added"
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-116"
    >
      Submit
    </span>
  </button>
</div>
`;

exports[`Renders correctly in every state without results 2`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiFormControl-root-98"
  >
    <div
      className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-110"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Candidate to be added"
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-116"
    >
      Submit
    </span>
  </button>
</div>
`;

exports[`Renders correctly in every state without results 3`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiFormControl-root-98"
  >
    <div
      className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-110"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Candidate to be added"
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-116"
    >
      Submit
    </span>
  </button>
</div>
`;

exports[`Renders correctly in every state without results 4`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiFormControl-root-98"
  >
    <div
      className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-110"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Candidate to be added"
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-116"
    >
      Submit
    </span>
  </button>
</div>
`;

exports[`Renders correctly with invalid address 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiFormControl-root-98"
  >
    <div
      className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-110"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Candidate to be added"
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-116"
    >
      Submit
    </span>
  </button>
</div>
`;
